lib/matplotlib/tests/test_legend.py

```diff
@@ -1450,3 +1450,38 @@ def test_boxplot_labels():
     handles, labels = ax.get_legend_handles_labels()
     assert len(handles) == 0
     assert len(labels) == 0
+
+
+def test_boxplot_legend_labels():
+    # Test that legend entries are generated when passing `label`.
+    np.random.seed(19680801)
+    data = np.random.random((10, 4))
+    fig, axs = plt.subplots(nrows=1, ncols=4)
+    legend_labels = ['A', 'B', 'C', 'D']
+
+    # Testing legend labels and patch passed to legend.
+    bp1 = axs[0].boxplot(data, patch_artist=True, label=legend_labels)
+    for i, v in enumerate(bp1['boxes']):
+        assert v.get_label() == legend_labels[i]
+    handles, labels = axs[0].get_legend_handles_labels()
+    assert labels == legend_labels
+    assert all(isinstance(h, mpl.patches.PathPatch) for h in handles)
+
+    # Testing legend without `box`.
+    bp2 = axs[1].boxplot(data, label=legend_labels, showbox=False)
+    # Without a box, The legend entries should be passed from the medians.
+    for i, v in enumerate(bp2['medians']):
+        assert v.get_label() == legend_labels[i]
+    handles, labels = axs[1].get_legend_handles_labels()
+    assert labels == legend_labels
+    assert all(isinstance(h, mpl.lines.Line2D) for h in handles)
+
+    # Testing legend with number of labels different from number of boxes.
+    with pytest.raises(ValueError, match='There must be an equal number'):
+        bp3 = axs[2].boxplot(data, label=legend_labels[:-1])
+
+    # Of 3 identical boxes, test that only the first box gets a label.
+    new_data = np.random.random((100, 3))
+    bp4 = axs[3].boxplot(new_data, label='A')
+    assert bp4['boxes'][0].get_label() == 'A' and all(x.get_label() != 'A'
+                                                      for x in bp4['boxes'][1:])
```

Comment:
I suggest to split the assertion instead of wrapping the line. Also `get_label() != 'A'` is not quite specific. Either test that it starts with an underscore (=convention to not include in the legend):

```suggestion
    assert bp4['boxes'][0].get_label() == 'A'
    assert all(x.get_label().startswith("_") for x in bp4['boxes'][1:])
```
or alternatively replace the second assertion by
```
handles, labels = axs[3].get_legend_handles_labels()
assert labels == ["A"]
```